optimization/+DoseConstraints/matRad_ObjectiveConstraint.m

```diff
@@ -0,0 +1,111 @@
+classdef matRad_ObjectiveConstraint < DoseConstraints.matRad_DoseConstraint
+    % matRad_ObjectiveConstraint implements a wrapper function that turns 
+    % dose objectives into constraints
+    %
+    % use log sum exp approximation, see appendix A in
+    % http://scitation.aip.org/content/aapm/journal/medphys/41/8/10.1118/1.4883837
+    %
+    % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    %
+    % Copyright 2020 the matRad development team. 
+    % 
+    % This file is part of the matRad project. It is subject to the license 
+    % terms in the LICENSE file found in the top-level directory of this 
+    % distribution and at https://github.com/e0404/matRad/LICENSES.txt. No part 
+    % of the matRad project, including this file, may be copied, modified, 
+    % propagated, or distributed except according to the terms contained in the 
+    % LICENSE file.
+    %
+    % %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    properties (Constant)
+        name = 'Objective Constraint';
+        parameterTypes = {'objFunc'};
+        parameterNames = {'f^{max}'};    
+    end
+
+    properties
+        objective;
+        parameters = {1e-5};
+        epsilon = 1e-3; %
```

Comment:
The epsilon is the slip factor? Maybe rename or comment it.
Does it make sense to have the f^max as a named parameter (in the parameter system) and the epsilon/slip not? shold both be just class properties / is there any scenario where using it from the GUI sensible?